options/secrets_test.go

```diff
@@ -0,0 +1,150 @@
+package options_test
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/coder/envbuilder/options"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetBuildSecrets(t *testing.T) {
+	tests := []struct {
+		name            string
+		envVars         map[string]string
+		expectedSecrets []string
+	}{
+		{
+			name:            "no secrets set",
+			envVars:         map[string]string{},
+			expectedSecrets: []string{},
+		},
+		{
+			name: "single secret",
+			envVars: map[string]string{
+				"ENVBUILDER_BUILD_SECRET_FOO": "bar",
+			},
+			expectedSecrets: []string{"FOO=bar"},
+		},
+		{
+			name: "multiple secrets",
+			envVars: map[string]string{
+				"ENVBUILDER_BUILD_SECRET_FOO": "bar",
+				"NOT_A_SECRET":                "baz",
+				"ENVBUILDER_BUILD_SECRET_BAZ": "qux",
+			},
+			expectedSecrets: []string{"FOO=bar", "BAZ=qux"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			envSnapshot := snapshotEnv()
+			t.Cleanup(func() { assertEnvUnchanged(t, envSnapshot) })
```

Comment:
This could lead to flakiness if someone is running these tests with a secret environment variable set in their shell env since we're using `os.Unsetenv` below and never restoring it.